### PR TITLE
Support binding to POCO when message body uses AMQP value

### LINKED ISSUE #31440: [Service Bus Extension] Binding to POCO does not handle non-standard message bodies
Similar to how the string binding logic will [inspect the message body](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToStringConverter.cs#L53) to determine whether it is a value body, we need to update the [POCO binding logic](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs) to have the same handling.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/31440

### CODE REVIEW

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs [hunk 1]
```diff
@@ -24,31 +25,56 @@ public MessageToPocoConverter(JsonSerializerSettings jsonSerializerSettings)
         public TElement Convert(ServiceBusReceivedMessage message)
         {
             // 1. If ContentType is "application/json" or binding to a JObject deserialize as JSON
-            // 2. If ContentType is not "application/json" attempt to deserialize using Message.GetBody, which will handle cases like XML object serialization
+            // 2. If ContentType is not "application/json" and it is a Data body message, attempt to deserialize using DataContractSerializer, which will handle cases like XML object serialization
             // 3. If this deserialization fails, do a final attempt at JSON deserialization to catch cases where the content type might be incorrect
 
             if (message.ContentType == ContentTypes.ApplicationJson ||
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs:31]: What happens if the POCO is `byte[]` or another form that implies "just give me the message body bytes"?

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs [hunk 2]
```diff
@@ -61,7 +87,7 @@ private TElement DeserializeJsonObject(ServiceBusReceivedMessage message)
                     CultureInfo.InvariantCulture,
 @"Binding parameters to complex objects (such as '{0}') uses Json.NET serialization or XML object serialization. 
  1. If ContentType is 'application/json' deserialize as JSON
- 2. If ContentType is not 'application/json' attempt to deserialize using Message.GetBody, which will handle cases like XML object serialization
+ 2. If ContentType is not 'application/json' and it is a Data body message, attempt to deserialize using DataContractSerializer, which will handle cases like XML object serialization
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs:90]: Should we consider linking out to something AMQP here?  I suspect "data body message" isn't going to mean something to a lot of people.

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs:31]: Binding to a byte array would not result in the Poco converter being executed. Instead, it is handled here - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToByteArrayConverter.cs

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs:90]: I think we should add this scenario to our troubleshooting guide and link to that.

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs:90]: Will need to do a separate PR once the updated TSG is merged.

**Pull request merged**